tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolBIRTest.java

```diff
@@ -296,4 +297,29 @@ public int hashCode() {
             return Objects.hash(this.name, this.kind);
         }
     }
+
+    @DataProvider(name = "hasDefaultTestProvider")
+    public Object[][] hasDefaultTest() {
+        return new Object[][]{
+                {21, 6, null},
+                {22, 6, true},
+                {23, 6, false}
+        };
+    }
+
+    @Test(dataProvider = "hasDefaultTestProvider")
+    public void testHasDefaultValue(int line, int col, Object hasDefault) {
```

Comment:
Minor thing, `Boolean` can be used here instead of `Object`

Reply:
In addition to boolean values we are handling null values as well (when the class field is not public),
Because of the reason I have used `'Object'` as the function parameter type.

Reply:
Updated in 118e6d43ebcd4bef45826d3e7ff54167b80e4ae5